Packages and features documentation in the readme (#362)
swent-sp-2024-event-radar/event-radar-app: README.md

# event-radar-app

> Holds all the code for the Android app

Reviewer: ```suggestion
Holds all the code for the Android app, Event Radar (see the demo).
```

TODO: Add demo

## Overview of functionalities

Event Radar is an innovative app designed to effortlessly facilitate connections among individuals by streamlining event discovery, registration, and networking in one place. It allows users to discover and register for upcoming events in their local area, enabling them to connect seamlessly with friends and new acquaintances they meet during the event. Event hosts can easily create and manage events while leveraging the app's unique QR code system to scan tickets and easily interact with event attendees.

### Features

- **Event Discovery:** Effortlessly find events based on your interests and location and view their position on the map.

- **Easy Registration:** Register for events with just a few taps.

- **Networking:** Connect with new friends and save their contact information by scanning QR codes.

- **Messaging:** Continue the conversation with new acquaintances after the event.

- **Event Hosting:** Create and manage events with robust tools for ticketing and attendee tracking.

- **Quick verification:** Scan attendees' QR codes to confirm their presence on the event's participant list.

### Testing

> The code is thoroughly tested, achieving 88% line coverage across our features with several hundred tests. Each screen displayed to users is covered by an Android UI test, ensuring the interface works correctly. Additionally, numerous unit tests verify the functionality of individual components. We also conduct end-to-end tests to ensure a smooth and accurate user flow, providing an optimal user experience.

Reviewer: The 88% might change - should update it later. Alternatively, say above 80% coverage.

### Running the App

> Download the built APK on your android phone from the release branch M4.

Reviewer: ```suggestion
Download the built debug APK on your android phone from the release branch M4.
```

## Packages structure

> Here is a description of the more important packages in the main folder. We have clearly separated fronted Ui from backend functionalities and data interactions.

Reviewer: ```suggestion
Here is a description of the most important packages in the main folder. We have clearly separated frontend Ui from backend functionalities and data interactions, following Model View ViewModel (MVVM) architecture.
```

- **Model** holds all the model for our app such as event, user, messages and the differents repositories to interact with the database

>   - **di** is used to store the setup files for data dependencies injection

Reviewer: ```suggestion
  - **di** is used to store the setup files for data dependencies injection using Hilt. 
```

  - **repository** holds the repositories used to bridge between the data source and the rest of the app. It contains repositories for : event, users, messages, and location. In our case we are working the Firebase and Nominatim.

> - **Ui** sort all the kotlin files containing our composable function to display the ui. Our main screens (brow)

Reviewer: ```suggestion
- **Ui** sort all the kotlin files containing our composable function to display the ui. The ui is built using Jetpack Compose Material3. Our main screens (brow)
```

> - **Ui** sort all the kotlin files containing our composable function to display the ui. Our main screens (brow)

Reviewer: Also, I think this sentence is unfinished?
`Our main screens (brow)`

Author: oops, indeed, thank you !

- **ViewModel** contains all our viewModels used in the app. ViewModels manage the data for the UI components, handling logic and surviving configuration changes, based on the Model View ViewModel design. 

> 

Reviewer: We also have a "util" folder. Perhaps add what that means too.

Author: I didn't mentioned it because it isn't really used. Only one file with one function for toast message, so I didn't includ it in the "most important packages"

Reviewer: Noted, sounds good 👌
